op-e2e/actions/l2_engine.go

```diff
@@ -176,12 +186,22 @@ func (e *L2Engine) ActL2IncludeTx(from common.Address) Action {
 			return
 		}
 
+		// Wait for the tx to be in the pending tx queue
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+		err := wait.For(ctx, 2*time.Second, func() (bool, error) {
+			i := e.engineApi.PendingIndices(from)
+			txs, _ := e.eth.TxPool().ContentFrom(from)
+			return uint64(len(txs)) > i, nil
+		})
+		require.NoError(t, err, "wait for tx to be in pending tx queue")
+
 		i := e.engineApi.PendingIndices(from)
 		txs, q := e.eth.TxPool().ContentFrom(from)
 		require.Greaterf(t, uint64(len(txs)), i,
 			"no pending txs from %s, and have %d unprocessable queued txs from this account", from, len(q))
```

Comment:
We should combine the waiting and "getting" of values here so that when the waiter returns we use the values it just validated match.  It _shouldn't_ matter but this code can have the tx pool change again between when the waiter completes and the actual get which causes the test to fail again.